### PR TITLE
Action Enumerators Context

### PR DESCRIPTION
**Pull request by ThyWoof**
Adding this initially as a tool for modders to get logging on Battle Service methods called from ActionMagicEffect and ActionAttack;

These patch the constructor that creates the internal class returned from these yield calls. In practice they work as Prefix to IEnumerator Method();

Can be used to get context on specific steps or change the call parameters as required;

### CODE REVIEW

**Comment by ThyWoof** [comment]
image

image

**Comment by ThyWoof** [comment]
image
multiple targets on spell

**Review by PhilPJL — COMMENTED** [review]
I'm going to need more time to understand what you're doing :)

The code definitely looks good, and produces some useful logging. Should they be warnings or info?

I haven't tried this but is an alternative less intrusive way to achieve the same thing to do something like this?

image

File: SolastaCommunityExpansion/Settings.cs [hunk 1]
```diff
@@ -27,6 +27,8 @@ public class Settings : UnityModManager.ModSettings
         // Diagnostics
         //
 
+        public bool EnableBattleServiceEventLogs { get; set; } = true;
```
Comment (PhilPJL) [inline SolastaCommunityExpansion/Settings.cs:30]: Should we default this to false?

**Pull request closed without merging**